lib/Installation/Warnings/WarningsController.pm

```diff
@@ -0,0 +1,49 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: The class introduces business actions for Warning Popups
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package Installation::Warnings::WarningsController;
+use strict;
+use warnings;
+
+use strict;
+use warnings;
+use YuiRestClient;
+use Installation::Warnings::ConfirmationWarningRichText;
+
+sub new {
```

Comment:
do we really need this controller? We could have some controller named for example yast_initialization or installer_initialization (or similar) which would use this warning, because we always use this as part of the test.
what about this way? https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/7d6fdcdc3b9b6917689b61e29856998af8830cdb/tests/installation/authentication/root_simple_pwd.pm#L27
Because I'm not sure how we guarantee that the warning is there, does it work in other archs?

Reply:
I would go with this controller, as I would like to experiment with this approach. It is similar to what we have in NavigationController, which seems to be working good and saved us a lot of potential duplication.

Here I see two improvements:
1. When popup is shown, LibyuiClient see it as the topmost page and does not see the page under it. So, we anyways can guarantee that the warning is there only by getting some its elements (e.g. label or warning text). So, with the approach of having WarningController, nothing will be changed on this matter.
2. Warnings may appear on any page, also there may be plenty of variants of the warning. e.g. like we have for Create User Page. We can have only `warning->get_text()` and then compare the received text with the expected one, provided in test module. So, no duplication, everything related to warnings will be in one controller, like we have for NavigationController.